### ISSUE TITLE
Treeshaking trying to evaluate non-constant expressions

### ISSUE THREAD

**Issue by lishid** [rollup/rollup#2651]
- Rollup Version: 1.1.0
- Operating System (or Browser): Windows 10 x64
- Node Version: 11.6.0

### How Do We Reproduce?

```
if ({}.hasOwnProperty) {
  console.log(true);
} else {
  console.log(false);
}
```

[REPL link](https://rollupjs.org/repl?version=1.1.0&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmlmJTIwKCU3QiU3RC5oYXNPd25Qcm9wZXJ0eSklMjAlN0IlNUNuJTIwJTIwY29uc29sZS5sb2codHJ1ZSklM0IlNUNuJTdEJTIwZWxzZSUyMCU3QiU1Q24lMjAlMjBjb25zb2xlLmxvZyhmYWxzZSklM0IlNUNuJTdEJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyY2pzJTIyJTJDJTIybmFtZSUyMiUzQSUyMm15QnVuZGxlJTIyJTJDJTIyYW1kJTIyJTNBJTdCJTIyaWQlMjIlM0ElMjIlMjIlN0QlMkMlMjJnbG9iYWxzJTIyJTNBJTdCJTdEJTdEJTJDJTIyZXhhbXBsZSUyMiUzQW51bGwlN0Q=)

### Expected Behavior

Output identical/similar to input

### Actual Behavior

```
'use strict';

{
  console.log(false);
}
```

**Comment by lukastaegert** [comment]
Thanks for spotting this, fix at #2652.

**Comment by lishid** [comment]
Thanks for the speedy fix!
